packages/web-app-files/tests/unit/helpers/share/link.spec.ts

```diff
@@ -0,0 +1,118 @@
+import { createQuicklink, CreateQuicklink } from '../../../../src/helpers/share/link'
+import { DateTime } from 'luxon'
+import { Store } from 'vuex'
+import { clientService } from 'web-pkg/src/services'
+import { useClipboard } from '@vueuse/core'
+import { Ability } from 'web-pkg'
+import { mock } from 'jest-mock-extended'
+
+jest.mock('@vueuse/core', () => ({
+  useClipboard: jest.fn().mockReturnValue({ copy: jest.fn() })
+}))
+
+const mockStore = {
+  state: {
+    user: {
+      capabilities: {
+        files_sharing: {
+          quickLink: {
+            default_role: 'viewer'
+          },
+          public: {
+            expire_date: {
+              enforced: true,
+              days: 5
+            }
+          }
+        }
+      }
+    }
+  },
+  dispatch: jest.fn(() => Promise.resolve({ url: '' }))
+}
+
+const mockResource = {
+  fileId: '1234',
+  path: '/path/to/file'
+}
+
+const getAbilityMock = (hasPermission) => mock<Ability>({ can: () => hasPermission })
+
+describe('createQuicklink', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
```

Comment:
We do this automatically, no need to call is explicitly.

```suggestion
  beforeEach(() => {
    jest.clearAllMocks()
  })
```